--- tengen-0.1.0/lib/tengen/game.rb	2004-12-30 00:36:38.000000000 +0100
+++ ../game.rb	2009-04-03 11:08:05.000000000 +0200
@@ -32,10 +32,10 @@
 
             def initialize
                 super
-                @notify_genmove = Mutex.new.lock
+                @notify_genmove = Queue.new
                 Thread.start do
                     loop do
-                        @notify_genmove.lock
+                        @notify_genmove.pop
                         Tengen.debug("Executing genmove for " \
                                      "#{@genmove_requester}.")
                         @genmove_requester.do_genmove
@@ -46,7 +46,7 @@
             def request_genmove(player)
                 @genmove_requester = player
                 Tengen.debug("Requesting genmove for #{player}.")
-                @notify_genmove.unlock
+                @notify_genmove.push(:dummy)
             end
 
             def reset!
